src/java.base/share/classes/java/lang/classfile/instruction/CharacterRange.java

```diff
@@ -40,7 +41,29 @@
  * {@link CharacterRangeTableAttribute}.  Delivered as a {@link CodeElement}
  * during traversal of the elements of a {@link CodeModel}, according to
  * the setting of the {@link ClassFile.DebugElementsOption} option.
+ * <p>
+ * Conceptually, a character range entry is a record:
+ * {@snippet lang=text :
+ * // @link region=0 substring="CharacterRange" target="#of"
+ * // @link region=1 substring="Label startScope" target="#startScope"
+ * // @link region=2 substring="Label endScope" target="#endScope"
+ * // @link region=3 substring="int characterRangeStart" target="#characterRangeStart"
+ * // @link region=4 substring="int characterRangeEnd" target="#characterRangeEnd"
+ * // @link substring="int flags" target="#flags()" :
+ * CharacterRange(Label startScope, Label endScope, int characterRangeStart, int characterRangeEnd, int flags)
+ * // @end region=0
+ * // @end region=1
+ * // @end region=2
+ * // @end region=3
+ * // @end region=4
+ * }
+ * <p>
+ * Physically, a character range has the same structure; it is modeled by a
+ * {@link CharacterRangeInfo}.
```

Comment:
I'm not sure how to explain "Physically" and "it is modeled by".
If "Physically" means specific byte serialization, then it is an entry in a CRT attribute with BCIs instead of labels.

"it is modeled by a CharacterRangeInfo" seems to me very confusing as this class (CharacterRange) is the primary model class. 

Maybe we should explain the duplicity of some classfile structures accessible as attributes internal raw structures (CharacterRangeInfo) versus preferred primary models as labeled pseudo instructions (CharacterRange).

Reply:
Done. I have simplified the record model and added dedicated info about CRI being independent from Code so it's used in the attribute. Same for LineNumber etc.